react router to load recipe page

wangfowen opened the pull request:
- set up a dev express server to locally test routing
- changed rendering of recipe component to happen on its own route
- updated react to use latest react router
- RecipeList + RecipesContainer => RecipesList
- RecipeContainer => RecipeLink

wangfowen committed:
react router to load recipe page

- set up a dev express server to locally test routing
- changed rendering of recipe component to happen on its own route
- updated react to use latest react router
- RecipeList + RecipesContainer => RecipesList
- RecipeContainer => RecipeLink

wangfowen commented:
oh, each recipe page should also have a load recipe button (temporarily, trigger on page load when it's confirmed working) to call the action for if we load that page directly. clicking the link from index page triggers the action already
recipe page just loads the generic recipe component right now, haven't figured out how to get it to load the state for that specific recipe after navigation

sherbondy commented:
Might be better to just use express's built-in static file middleware?
https://expressjs.com/en/starter/static-files.html

wangfowen committed:
adjustments to server

sherbondy committed:
seemingly working setup for dev and production.

- Had to shuffle around some directory references, /dist/, etc.
- Reintroduced Owen's catchall route for index.html
  - Should maybe be explicit about which routes are supposed to redirect to /index
  - ... "isomorphic" router so we don't define the same logic in two places?
- Little peachworks API proxy module factored out so server is less messy

sherbondy committed:
had to move some deps out of dev for production now build to work.

sherbondy commented:
Okay, I think it works alright now.
Re-added your catch-all route.

There seemed to just be issues with relative urls and /dist/bundle.js vs /bundle.js, had to make consistent across dev & prod where the file paths ended up.

There was also an issue where production env couldn't build the client js bundle because a lot of the babel stuff was specified as dev dependencies but required for the prod build.
